tests/test_flagsmith.py

```diff
@@ -172,6 +173,77 @@ def test_get_identity_flags_uses_local_environment_when_available(
     assert identity_flags[0].value == feature_state.get_value()
 
 
+@responses.activate()
+def test_get_identity_flags__transient_identity__calls_expected(
+    flagsmith: Flagsmith,
+    identities_json: str,
+    environment_model: EnvironmentModel,
+    mocker: MockerFixture,
+) -> None:
+    # Given
+    responses.add(
+        method="POST",
+        url=flagsmith.identities_url,
+        body=identities_json,
+        match=[matchers.json_params_matcher({"transient": True}, strict_match=False)],
+    )
+    flagsmith._environment = environment_model
+    flagsmith.enable_local_evaluation = True
+    mock_engine = mocker.patch("flagsmith.flagsmith.engine")
+
+    feature_state = FeatureStateModel(
+        feature=FeatureModel(id=1, name="some_feature", type="STANDARD"),
+        enabled=True,
+        featurestate_uuid=str(uuid.uuid4()),
+    )
+    mock_engine.get_identity_feature_states.return_value = [feature_state]
+
+    # When & Then
+    flagsmith.get_identity_flags(
+        "identifier",
+        traits={"some_trait": "some_value"},
+        transient=True,
+    ).all_flags()
```

Comment:
Hmmm... at first glance, it looks like this isn't actually doing anything, but I think maybe the `match` argument to the `responses.add()` call is doing some part of this at least? 

I definitely think that we need to document better what's happening as it's odd to me to not see any assertion here.

Reply:
Responses' matchers are documented [here](https://github.com/getsentry/responses?tab=readme-ov-file#id17).

Lax matching did not work as I expected though. Fixed the tests to match full request bodies strictly.